ygot/struct_validation_map.go

```diff
@@ -475,14 +475,33 @@ func MergeJSON(a, b map[string]interface{}) (map[string]interface{}, error) {
 	return o, nil
 }
 
+// MergeOpt is an interface that is implemented by the options to the MergeStructInto
+// function. It allows user specified options to be propagated to the merge
+// method.
+type MergeOpt interface {
+	// IsMergeOpt is a marker method for each MergeOpt.
+	IsMergeOpt()
+}
+
+// DiffPathOpt is a DiffOpt that allows control of the path behaviour of the
+// Diff function.
+type MergeStructOpt struct {
```

Comment:
I'd suggest that we make this a more specific option - since right now I can't see how there would be >1 different option -- it might be better to simply make this be:

```golang
type MergeOverwriteExistingFields struct { }
```

and then simply use the presence of this option as indicating that overwriting should be enabled.